pkg/core/native/native_gas.go

```diff
@@ -129,6 +129,15 @@ func (g *GAS) OnPersist(ic *interop.Context) error {
 
 // PostPersist implements the Contract interface.
 func (g *GAS) PostPersist(ic *interop.Context) error {
+	for _, tx := range ic.Block.Transactions {
+		attrs := tx.GetAttributes(transaction.RefundableSystemFeeT)
+		if len(attrs) != 0 {
+			consumed := ic.TxesConsumed[tx.Hash()]
+			if consumed < tx.SystemFee {
+				g.mint(ic, tx.Sender(), big.NewInt(tx.SystemFee-consumed), false)
```

Comment:
I'd still check `tx.Sender()` for being a contract address here. Contract addresses are predictable, one can precalculate it, send some GAS to it, use it as a sender in a transaction and deploy a contract in the same block (making regular address a contract address).

Reply:
Or actually this can't happen because to be a sender an account should not just have some GAS, but also pass a witness check. Contract scripts are very specific and not supposed to ever pass it if they're not deployed. When deployed, a previous check is already sufficient.